go-controller/pkg/node/management-port.go

```diff
@@ -109,6 +113,31 @@ func (mp *managementPort) Create(nodeAnnotator kube.Annotator, waiter *startupWa
 	return cfg, nil
 }
 
+func (mp *managementPort) checkExistingMgmtPort() error {
+	_, stderr, err := util.RunOVSVsctl("list", "Interface", types.K8sMgmtIntfName)
+	if err != nil {
+		if !strings.Contains(stderr, "no row \""+types.K8sMgmtIntfName+"\" in table") {
+			return fmt.Errorf("failed to lookup OVS port: %s", stderr)
+		}
+		// Not found on the bridge. But could be that interface with the same name exists
```

Comment:
> Could this legacy interface be representor port?

Have no idea. But AFAIU from comparing with MgmtNetdev and MgmtRep it is only OVS internal port.

> How do we know only platform mgmt interface needs to be handled here?

What else?

Reply:
> > Could this legacy interface be representor port?
> 
> Have no idea. But AFAIU from comparing with MgmtNetdev and MgmtRep it is only OVS internal port.

From HW full mode to OVS internal port:
1) the mgmt port name was `ovn-k8s-mp0_0` in HW full mode
2) `ovs-vsctl list interface ovn-k8s-mp0` return `no row in the table`
3) it goes to unconfigurePlatformInterface with `ovn-k8s-mp0` which renames the VF netdev to `ovn-k8s-mp0-ddmmyyHHMMSS` and return success
4) this leaves `ovn-k8s-mp0_0` remain in `br-int` bridge

> 
> > How do we know only platform mgmt interface needs to be handled here?
> 
> What else?

The representor port left in step 4, should it be renamed or released?

Reply:
> > > Could this legacy interface be representor port?
> > 
> > 
> > Have no idea. But AFAIU from comparing with MgmtNetdev and MgmtRep it is only OVS internal port.
> 
> From HW full mode to OVS internal port:
> 
> 1. the mgmt port name was `ovn-k8s-mp0_0` in HW full mode
> 2. `ovs-vsctl list interface ovn-k8s-mp0` return `no row in the table`
> 3. it goes to unconfigurePlatformInterface with `ovn-k8s-mp0` which renames the VF netdev to `ovn-k8s-mp0-ddmmyyHHMMSS` and return success
> 4. this leaves `ovn-k8s-mp0_0` remain in `br-int` bridge
> 

Ok, lets clarify. You are talking about DPU full representor. Not about `legacyMgmtIntfName := util.GetLegacyK8sMgmtIntfName(mp.nodeName)`.

> > > How do we know only platform mgmt interface needs to be handled here?
> > 
> > What else?
> 
> The representor port left in step 4, should it be renamed or released?

Renamed and removed, I suppose, because of flows, You mentioned below.

Reply:
You know. I had a talk to team mate and in multicluster scenario we actually have two representors on the br-int and it works.

Reply:
> > > > Could this legacy interface be representor port?
> > > 
> > > 
> > > Have no idea. But AFAIU from comparing with MgmtNetdev and MgmtRep it is only OVS internal port.
> > 
> > 
> > From HW full mode to OVS internal port:
> > 
> > 1. the mgmt port name was `ovn-k8s-mp0_0` in HW full mode
> > 2. `ovs-vsctl list interface ovn-k8s-mp0` return `no row in the table`
> > 3. it goes to unconfigurePlatformInterface with `ovn-k8s-mp0` which renames the VF netdev to `ovn-k8s-mp0-ddmmyyHHMMSS` and return success
> > 4. this leaves `ovn-k8s-mp0_0` remain in `br-int` bridge
> 
> Ok, lets clarify. You are talking about DPU full representor. Not about `legacyMgmtIntfName := util.GetLegacyK8sMgmtIntfName(mp.nodeName)`.
> 
yes, HW full offload, not legacy k8s mgmtn interface name.

Reply:
Could you elaborate on the multicluster scenario? Does it address the remaining ovn-k8s-mp0_0 issue from HW full mode to OVS internal port in single cluster?

Reply:
In Multicluster two instances of ovnkube-node are running: in DPU mode and full mode. To make them not fight over all resources, they are separated, including management ports, so there are two management ports with different names added to br-int. AFAIK, there is no changes to separate flows.